db/migrate/20170927135007_migrate_dialog_field_associations_to_use_new_relationship.rb

```diff
@@ -0,0 +1,70 @@
+    include MigrationStubHelper # NOTE: Must be included after other mixins
+  end
+  class DialogField < ActiveRecord::Base
+    include MigrationStubHelper # NOTE: Must be included after other mixins
+  end
+  class Dialog < ActiveRecord::Base
+    include MigrationStubHelper # NOTE: Must be included after other mixins
+  end
+  class DialogTab < ActiveRecord::Base
+    include MigrationStubHelper # NOTE: Must be included after other mixins
+  end
+  class DialogGroup < ActiveRecord::Base
+    include MigrationStubHelper # NOTE: Must be included after other mixins
+  end
+
+  def up
+    say_with_time("Migrating existing dialog field associations to new relationship") do
+      dialog_fields_with_associations = DialogField.all.select { |df| df.auto_refresh || df.trigger_auto_refresh }
+      dialog_with_fields = dialog_fields_with_associations_by_dialog(dialog_fields_with_associations)
+      dialog_with_fields.each do |_dialog, fields|
+        dialog_triggers = dialog_fields_with_trigger_auto_refresh(fields).sort_by! { |trigger| trigger[:position] }
+        dialog_responders = dialog_fields_with_auto_refresh(fields)
+        dialog_triggers.each_with_index do |trigger, index|
+          specific_responders = dialog_responders.select { |responder| responder_range(trigger, trigger[index + 1]).cover?(responder[:position]) }
+          set_associations(trigger, specific_responders)
+        end
+      end
+    end
+  end
+
+  def absolute_position(dialog_fields)
+    positions = []
+    dialog_fields.each do |f|
+      field_position = f.position
+      dialog_group_position = DialogGroup.find(f.dialog_group_id).position
+      dialog_tab_position = DialogTab.find(DialogGroup.find(f.dialog_group_id).dialog_tab_id).position
+      index = field_position + dialog_group_position * 100 + dialog_tab_position * 10_000
+      positions << {:id => f.id, :position => index}
+    end
+    positions
+  end
+
+  def dialog_fields_with_auto_refresh(dialog_fields_with_associations)
+    absolute_position(dialog_fields_with_associations.select { |df| df.auto_refresh == true })
+  end
+
+  def dialog_fields_with_trigger_auto_refresh(dialog_fields_with_associations)
+    absolute_position(dialog_fields_with_associations.select { |df| df.trigger_auto_refresh == true })
+  end
+
+  def responder_range(trigger_min, trigger_max)
+    min = trigger_min[:position] + 1
+    max = trigger_max.present? ? trigger_max[:position] - 1 : 10_000_000
+    (min..max)
+  end
+
+  def set_associations(trigger, specific_responders)
+    specific_responders.each { |responder| DialogFieldAssociation.create(:trigger_id => trigger[:id], :respond_id => responder[:id]) }
+  end
+
+  def dialog_fields_with_associations_by_dialog(dialog_fields_with_associations)
+    dialog_fields_with_associations.group_by { |df| Dialog.find(DialogTab.find(DialogGroup.find(DialogField.find(df.id).dialog_group_id).dialog_tab_id).dialog_id) }
+  end
+
+  def down
+    DialogFieldAssociation.delete_all
```

Comment:
Will the previous migration just drop the table?  If so, then there's no need for this at all.  (You could just put a comment stating that fact)